internal/policies/mount/mount_test.go

```diff
@@ -0,0 +1,212 @@
+package mount_test
+
+import (
+	"context"
+	"os"
+	"os/user"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/adsys/internal/policies/entry"
+	"github.com/ubuntu/adsys/internal/policies/mount"
+	"github.com/ubuntu/adsys/internal/testutils"
+)
+
+func TestNew(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		readOnlyPerm bool
+
+		wantErr bool
+	}{
+		"creates manager successfully": {},
+
+		"creation fails with invalid runDir permissions": {readOnlyPerm: true, wantErr: true},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			runDir := t.TempDir()
+			if tc.readOnlyPerm {
+				testutils.MakeReadOnly(t, runDir)
+			}
+
+			_, err := mount.New(runDir)
+			if tc.wantErr {
+				require.Error(t, err, "Expected an error when creating manager but got none.")
+				return
+			}
+			require.NoError(t, err, "Expected no error when creating manager but got one.")
+		})
+	}
+}
+
+func TestApplyPolicy(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		entries    []string
+		key        string
+		objectName string
+		isComputer bool
+
+		secondCall []string
+
+		// User specific
+		readOnlyUsersDir  bool
+		userReturnedUID   string
+		userReturnedGID   string
+		pathAlreadyExists bool
+
+		wantErr bool
+	}{
+		// USER // Single entry cases.
+
+		"successfully apply policy for entry with one value":        {},
+		"successfully apply policy for entry with multiple values":  {entries: []string{"entry with multiple values"}},
+		"successfully apply policy for entry with repeatead values": {entries: []string{"entry with repeatead values"}},
+
+		// USER // Special cases.
+		"successfully apply policy with anonymous values":                                   {entries: []string{"entry with anonymous tags"}},
+		"generates no file for errored entry":                                               {entries: []string{"errored entry"}},
+		"generates no file when trying to apply policy with user key and isComputer is set": {isComputer: true},
+
+		// USER // Badly formatted entries.
+		"successfully apply policy trimming whitespaces":           {entries: []string{"entry with linebreaks and spaces"}},
```

Comment:
This case has an empty golden file result, which is due to `entry with linebreaks and spaces` not existing in entries, so being nil.

I suggest you do a second round of looking at all the goden files content and that they makes sense compared to the input/test case :)